crates/iota-data-ingestion-core/README.md

```diff
@@ -0,0 +1,352 @@
+
+#### Local
+
+The local checkpoint fetcher is ideal when the indexer and a Full Node run on the same machine. By configuring the Full Node to save generated checkpoints to a local directory, `iota-data-ingestion-core` can monitor this directory for changes using an inotify-based mechanism. This minimizes ingestion latency, as checkpoints are processed immediately after they are generated by the Full Node.
+
+#### Remote
+
+The remote checkpoint fetcher is useful when the indexer and Full Node reside on separate machines. The Full Node can be configured to expose a REST API for checkpoint retrieval.
+
+#### Hybrid
+
+A hybrid configuration combines local and remote stores, providing a fallback mechanism for continuous data ingestion. The framework prioritizes locally available checkpoint data. This mode is beneficial when transitioning to using a local Full Node for data ingestion while needing to backfill historical data or provide failover capabilities.
+
+## Detailed API Functionality
+
+To fully understand the inner workings and configuration options of `iota-data-ingestion-core`, a closer look at its API is necessary. While the `Worker` trait dictates the processing and storage of `CheckpointData`, the `IndexerExecutor` is responsible for orchestrating the workers and implementing the core ingestion logic. Before diving into the details of the `IndexerExecutor`, we will first analyze the components it relies on.
+
+### Checkpoint Reader
+
+The `CheckpointReader` actor is responsible for reading and managing `CheckpointData` from multiple sources, including local files and remote storage. In addition to fetching new checkpoints, it tracks which checkpoints have been processed, performs garbage collection by deleting processed checkpoint files from local storage, and enforces a memory limit during batch processing to prevent out-of-memory (OOM) conditions. The `CheckpointReader` is instantiated using the `initialize(..)` method.
+
+```rust
+pub fn initialize(
+    // The directory on which checkpoints will be read
+    path: PathBuf,
+    // From which checkpoint sequence number to start fetching new checkpoints
+    starting_checkpoint_number: CheckpointSequenceNumber,
+    // Provide a remote URL to download checkpoints
+    remote_store_url: Option<String>,
+    // Any config key-value pair for the Remote fetcher
+    remote_store_options: Vec<(String, String)>,
+    // Customize and optimize the Checkpoint fetch behavior
+    options: ReaderOptions,
+) -> (
+    Self,
+    // Channel the Workers will listen and process & store checkpoints
+    mpsc::Receiver<CheckpointData>,
+    // Channel on which workers will notify GC of the checkpoint
+    mpsc::Sender<CheckpointSequenceNumber>,
+    // Graceful shutdown
+    oneshot::Sender<()>,
```

Comment:
maybe something to update considering the `CancelationToken` change?

Reply:
You're right, for now I was thinking to leave it as it is, a new [requirement](https://github.com/iotaledger/rebased-stardust-indexer/issues/11#issuecomment-2539307848) for the `cusotm-indexer` needs some investigation, and this could affect the implementation of the `CancelationToken` on the data ingestion logic. Once I'll have an answer on the requirement I'll change the README accordingly, int his issue or in a next one, what do you think?